crates/primitives/src/transaction/signature.rs

```diff
@@ -99,8 +99,18 @@ impl Signature {
         // errors and we care only if recovery is passing or not.
         secp256k1::recover(&sig, hash.as_fixed_bytes()).ok()
     }
-}
 
+    /// Turn this signature into its byte
+    /// (hex) representation.
+    pub fn to_bytes(&self) -> Bytes {
+        let mut sig = [0u8; 65];
+        sig[..32].copy_from_slice(&self.r.to_be_bytes::<32>());
+        sig[32..64].copy_from_slice(&self.s.to_be_bytes::<32>());
+        let v = u8::from(self.odd_y_parity) + 27;
```

Comment:
Do we need to require passing a chain id here by any chance, to respect EIP-155? It is a bit awkward because I think EIP-155 means the signature may end up being larger than 65 bytes if the `v` value exceeds 255 for a large chain id? We've had this conversation before in ethers cc @mattsse

Reply:
in ethers we have the issue that we store the `v: u64` and assumes:

```
        // The u64 to u8 cast is safe because `sig.v` can only ever be 27 or 28
        // here. Regarding EIP-155, the modification to `v` happens during tx
        // creation only _after_ the transaction is signed using
        // `ethers_signers::to_eip155_v`.
        sig[64] = src.v as u8;
```

which is a bit unsafe.

https://github.com/gakonst/ethers-rs/blob/2cead3e11160a48b403d1334eab329d38f10b4fa/ethers-core/src/types/signature.rs#L235-L239

but if properly used the v values is set to  (0|1) + 27:

```
        let v = u8::from(recoverable_sig.recovery_id()) as u64 + 27;
```


https://github.com/gakonst/ethers-rs/blob/2cead3e11160a48b403d1334eab329d38f10b4fa/ethers-signers/src/wallet/mod.rs#L156-L156

we already have (0|1) for `odd_y_parity` so we need to add the 27 when setting the v byte